### ISSUE TITLE
Add auxiliary routes to Router

### ISSUE THREAD

**Issue by sskieller** [dotnet/aspnetcore#43654]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe the problem.

I have a situation where I would really like to be able to open a dialog without having to go to a specific page. 

So for example,
if I opened the link /calendar/view(jobedit:{id}) it would open the modal for the createJob even though I was in the calendar view. and then give me a LINK like that, which can be opened from another application.

If I wanted to see the job, or send somebody else the jobid, I should be able to send them whatever route like /containers/list(jobedit:{id}) and then it would keep the containers/list open while opening the viewJob modal on top. 

### Describe the solution you'd like

Basically allow auxiliary routing where you can route to multiple components independent of each other.

I have considered relative paths, where the dialog component only opens on a certain path, but relative paths isn't a thing it seems in the blazor Router

I have considered creating a route for each route possible, but this approach is very clunky.
/job/grid/default
/job/grid/default/job/{id}
/job/grid/global
/job/grid/global/job/{id}
/calendar/view
/calendar/view/job/{id}
Instead of this, it would be nice to just have 
/job/grid/default
/job/grid/global
/calendar/view
and then the (jobedit:{id}) would just catch all if that was added

like /job/grid/global(jobedit:{id})

**Comment by javiercn** [comment]
Duplicate of https://github.com/dotnet/aspnetcore/issues/11212
